arch/xtensa/include/syscall.h

```diff
@@ -41,6 +64,194 @@
  * Inline functions
  ****************************************************************************/
 
+#ifndef __ASSEMBLY__
+
+/* SVC call with SYS_ call number and no parameters */
+
+static inline uintptr_t sys_call0(unsigned int nbr)
+{
+  register long reg0 __asm__("a2") = (long)(nbr);
+
+  __asm__ __volatile__
+  (
+    "isync\n"
+    "dsync\n"
+    "esync\n"
+    "memw\n"
+    "movi a3, %1\n"
+    "wsr a3, intset\n"
```

Comment:
Here we are triggering a software interrupt, the cause will be interruptlevel1 or syscall?

Reply:
Yes, we use software interrupt  instead of syscall.

As I mentioned in title.

`syscall` instruction has the same level (interruptlevel1) with other common interrupts (interruptlevel1).

We must use a higher interrupt level that can generate interrupt when `enter_critical_section()`

The sw interrupt in our hardware design, is hardwired to interrupt level2. 

So We  can use it.

Reply:
The premise is that  in hardware design,  we have a software interrupt in higher level.

This is not the very common solution

But I can't find another method to solve this problem....

Reply:
@Ouss4 the key point is that the critical section also mask syscall, do you have better method to fix the problem?

Reply:
@zhuyanlinzyl the namings are a bit confusing here.  The title says "Use SVC call in context swtich", SVC is an ARM thing, which I thought was meant to say `syscall`.  In your last PR you fixed the entry into the exception caused by `syscall` so I thought this is the continuation of that work.

Anyway, back to the software interrupt.  So you have a software interrupt hardwired to interrupt level2 and it's attached to the `xtensa_svcall` ISR. I would suggest we fix the naming later, and call it `swint` as we do in RISC-V and others.

@xiaoxiang781216 @zhuyanlinzyl what level is your timer interrupt? It has to be smaller than or same level as your software interrupt level for this to work?  Otherwise the software interrupt will still be masked.

Reply:
@Ouss4  Sorry for last relay, We use sw interrupt in level 3. and timer interrupt in level2.

Yes.  A bit confused here, as I first want to use syscall, So I fixed syscall handler error, but  found I can't use it later.